fmt/format.h

```diff
@@ -748,7 +758,7 @@ void MemoryBuffer<T, SIZE, Allocator>::grow(std::size_t size) {
   std::size_t new_capacity = this->capacity_ + this->capacity_ / 2;
   if (size > new_capacity)
       new_capacity = size;
-  T *new_ptr = this->allocate(new_capacity);
```

Comment:
See http://en.cppreference.com/w/cpp/memory/allocator/allocate
Second argument is defaulted to literal 0 and causes a warning when -Wzero-as-null-pointer-constant diagnostic enabled.
I presume compilation error happens in mock-allocator.h?
If so, just adding a dummy parameter to its allocate() will fix that (and also will make that mock more conformant).